.github/workflows/main-build-java.yml

```diff
@@ -0,0 +1,199 @@
+name: Java Layer Integration Test
+on:
+  workflow_call:
+    inputs:
+      caller-workflow-name:
+        required: true
+        type: string
+  workflow_dispatch:
+
+concurrency:
+  group: main-build-java-${{ github.ref_name }}
+  cancel-in-progress: true
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  integration-test:
+    runs-on: ubuntu-20.04
+    name: java-${{ matrix.sample-app }}-${{ matrix.instrumentation-type }}-${{ matrix.architecture }}-${{matrix.confmap}}
+    strategy:
+      fail-fast: false
+      matrix:
+        architecture: [ amd64, arm64 ]
+        runtime: [java17, java11, java8.al2]
+        sample-app: [ aws-sdk, okhttp ]
+        instrumentation-type: [ wrapper ]
+        confmap: [ "noop" ]
+        include:
```

Comment:
The include statement here was a bit hard to grok for me. Would be easier to invert this by adding all possible values in the the matrix key values, and then using [`exclude`](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#excluding-matrix-configurations) instead to remove combinations we don't want. Exclude could™ be easier to read because it only requires partial matches. 

It seems like most of our logic here is just excluding `okhttp` pairing with `agent` and `agent-confmap`

Reply:
yeah, we would also want to exclude `agent-conf map` to run on different runtimes of java  Since `agent` and `agent-confmap` are the same tests and confmap functionality tested in single runtime is good enough. i will update the PR.

Reply:
Ahh I looked into it seems like usage of include is better in this case. The agent-confmap will include conf map for http, https and s3. So even if we have to use exclude, unless we add `instrumentation-type: agent` in the matrix. we have to use include again.

correct me if my understanding is wrong.